### ISSUE TITLE
shouldn't this work?

### ISSUE THREAD

**Issue by Jeremy5909** [vivaansinghvi07/dreamberd-interpreter#14]
```c++
const var text = "ben"!
text[0.5] = "e"!
print text!
```

instead i get error
> TypeError: slice indices must be integers or None or have an __index__ method

**Comment by vivaansinghvi07** [comment]
Yes, I remember patching something like this for lists but I guess I forgot strings. Thanks!

**Comment by Jeremy5909** [comment]
well now it works but you cant do text[0.5] or anything, it just goes to one. Still good tho and this might he hard to implement

**Comment by vivaansinghvi07** [comment]
What do you mean by goes to one?

**Comment by Jeremy5909** [comment]
Like it seems once you do that code the text is 
b[-1]
e[0]
e[1]
n[2]

Instead of

b[-1]
e[0]
e[0.5]
n[1]
Because if i print text[1] it prints e and text[2] prints n

**Comment by vivaansinghvi07** [comment]
Ohhhh yeah that would be pretty annoying to implement, I’ll consider doing that in the future because I do think it’s definitely possible to.

**Comment by vivaansinghvi07** [comment]
I also don’t know if I should implement that behavior because it might make string parsing a nightmare 😂

**Comment by TodePond** [comment]
yeah the DB spec doesnt specify how this should work so feel free to do what you prefer :)

**Comment by CyberKite** [comment]
maybe float indexes could be done with heavy abuse of dictionaries?
